src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
 
-class DeliveryDrone {}
+    this.coords = {
+      x, y,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  getInfo() {
+    return [
+      `Robot: ${this.name}`,
+      `Chip version: ${this.chipVersion}`,
+      `Weight: ${this.weight}`,
+    ].join(', ');
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+  goRight(step = 1) {
+    this.coords.x += step;
```

Comment:
```suggestion
  goForward(step = 1) {
    this.coords.y += step;
  }
  
  goBack(step = 1) {
    this.coords.y -= step;
  }
  
  goRight(step = 1) {
    this.coords.x += step;
```
Add empty lines between methods to make your code easier to read